HikVisionSource/README.md

```diff
@@ -0,0 +1,203 @@
+         "DVRPassword": "pass2"
+      },
+   ],
+   "options": {
+      "maxActiveTasks": 3,
+      "maxQueuedTasks": 20
+   }
+}
+```
+
+### Configuration
+
+*   **sdkLogPath**: Optional. the folder where the HikVision sdk log files are going to be created , default value `c:\tmp\log`
+*   **DVRImageFolderPath**: Optional . The prefix of the file pattern to look for , if not set an empty string will be used as default value `c:/tmp/Thermo`
+*   **VantiqDocumentPath**: Optional. Location in Vantiq Document , default is `public/image`
+*   **VantiqResourcePath**: Optional. define the vantiq resource to be use for the upload (currently documents or images-`/resource/images`) default value : `/resource/documents`
+*   **cameras**: list of cameras as source for notifications
+
+### Options Available for camera
+
+for each camera the following properties must be included - all fieds are mandatory 
+
+*   **CameraId**: Camera Name to be use for identify event source .
+*   **Enable**: "True"|"False" , disabled camera will not included in the active camera list .  
+*   **DVRIP**: TCP address of the current camera
+*   **DVRPort**: TCP port of the ucrrent camera
+*   **DVRUserName**: username for authentication with the current camera
+*   **DVRPassword**: password for authentication with the current camera
+
+## Publish Statements
+
+Using Publish statements one can change the direction and zoom of the Hikvision camera. 
+
+``` 
+
+PROCEDURE TestHikvisionCommands()
+var r = {}
+var zoomIn = 11 
+var zoomOut = 12 
+var tiltUp = 21 
+var tiltDown = 22
+var start = 0 
+var stop = 1
+var PAN_RIGHT = 24 
+var PAN_LEFT = 23 
+publish {"cameraId":"c1","command":tiltDown,"state":start} to SOURCE HikVision1 using {body=r}
+```
+
+## Messages from the Source
+
+Messages that are sent to the source as Notifications from the camera. These are JSON objects with a similar format to the example above. The example below is the result of notification which include images as well. 
+
+## Running the example
+
+As noted above, the user must define the HikVision Source implementation in the Vantiq system. For an example of the definition, please see the `HikVisionImpl.json` file located in the `src/test/resources directory` .
+
+Additionally, an example project named `HikVisionExample.zip` can be found in the `src/test/resources directory` . 
+
+## Error Messages
+
+Query errors originating from the source will always have the code be the fully-qualified class name with a small descriptor 
+attached, and the message will include the exception causing it and the request that spawned it.
+
+The exception thrown by the HikVision Class will always be a VantiqHikVisionException. This is a wrapper around the traditional Exception, and contains the Error Message, and Error Code from the original Exception.
+
+## Licensing
+
+The source code uses the [MIT License](https://opensource.org/licenses/MIT).  
+
+HikariCP, okhttp3, log4j, and jackson-databind are licensed under
```

Comment:
You don't actually use HikariCP, so this should be removed.

Reply:
done